types/k6/browser/index.d.ts

```diff
@@ -936,6 +936,69 @@ export interface ConsoleMessage {
     type(): string;
 }
 
+/**
+ * {@link MetricMessage} objects are dispatched by page via the
+ * `page.on('metric')` event. For each metric that it measured and emitted for
+ * the page, k6 browser delivers it to the registered handlers.
+ *
+ * ```js
+ * // Listen for all metric messages in the page and call its tag method to
+ * // tag matching urls with the new tag name.
+ * page.on('metric', (metric) => {
+ *   metric.tag({
+ *     name: 'test',
+ *     matches: [
+ *       {url: /^https:\/\/test\.k6\.io\/\?q=[0-9a-z]+$/, method: 'GET'},
+ *     ]
+ *   });
+ * });
+ * ```
+ */
+export interface MetricMessage {
+    /**
+     * tag will match the given `tagMatch.matches` with the current metric's url
+     * and name tags. When a match is found it will use `tagMatch.name` to replace
+     * the existing url and name tag values.
+     * 
+     * Doing this will help group metrics with disparate url and name tags, which
+     * are in fact referencing the same resource, so that a correlation can be found
+     * over time and preventing high cardinality issues in the backend database that
+     * is storing the metrics.
```

Comment:
I would rephrase this to remove the explanation about k6 backend internals:

```suggestion
     * Doing this helps to group metrics with disparate url and name tags, which
     * are, in fact, referencing the same resource so that a correlation can be 
     * found over time to reduce the high cardinality of the metrics.
```

Reply:
Thanks for the suggestion! I used that and changed it a bit to:

```suggestion
     * Doing this helps to group metrics with disparate url and name tags, which
     * are, in fact, referencing the same resource so that a correlation can be 
     * found over time and to reduce the cardinality of the metrics.
```

e10c6f3a6ddbb72fb2bdfdeb7d621ad1cb3f8825